closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -0,0 +1,1044 @@
+        }
+
+        Map<String, String> propsMapping = new HashMap<>();
+
+        // JDK properties name as key, restricted security properties value as value.
+        propsMapping.put("jdk.tls.disabledNamedCurves", properties.jdkTlsDisabledNamedCurves);
+        propsMapping.put("jdk.tls.disabledAlgorithms", properties.jdkTlsDisabledAlgorithms);
+        propsMapping.put("jdk.tls.ephemeralDHKeySize", properties.jdkTlsDphemeralDHKeySize);
+        propsMapping.put("jdk.tls.legacyAlgorithms", properties.jdkTlsLegacyAlgorithms);
+        propsMapping.put("jdk.certpath.disabledAlgorithms", properties.jdkCertpathDisabledAlgorithms);
+        propsMapping.put("jdk.security.legacyAlgorithm", properties.jdkSecurityLegacyAlgorithm);
+
+        for (Map.Entry<String, String> entry : propsMapping.entrySet()) {
+            String jdkPropsName = entry.getKey();
+            String propsNewValue = entry.getValue();
+
+            String propsOldValue = props.getProperty(jdkPropsName);
+            if (isNullOrBlank(propsOldValue)) {
+                propsOldValue = "";
+            }
+
+            if (!isNullOrBlank(propsNewValue)) {
+                String values = isNullOrBlank(propsOldValue) ? propsNewValue : (propsOldValue + ", " + propsNewValue);
+                props.setProperty(jdkPropsName, values);
+                if (debug != null) {
+                    debug.println("Added restricted security properties, with property: " + jdkPropsName + " value: "
+                            + values);
+                }
+            }
+        }
+
+        // For keyStore and keystore.type, old value not needed, just set the new value.
+        String keyStoreType = properties.keyStoreType;
+        if (!isNullOrBlank(keyStoreType)) {
+            props.setProperty("keystore.type", keyStoreType);
+        }
+        String keyStore = properties.keyStore;
+        if (!isNullOrBlank(keyStore)) {
+            // SSL property "javax.net.ssl.keyStore" set at the JVM level via system properties.
+            System.setProperty("javax.net.ssl.keyStore", keyStore);
+        }
+    }
+
+    /**
+     * Check the restricted security properties
+     */
+    private static void restrictsCheck() {
+        // Check restricts object
+        if (restricts == null) {
+            printStackTraceAndExit("Restricted security property is null.");
+        }
+
+        // Check if the SunsetDate expired.
+        if (isPolicySunset(restricts.descSunsetDate)) {
+            printStackTraceAndExit("Restricted security policy expired.");
+        }
+
+        // Check secure random settings.
+        if (isNullOrBlank(restricts.jdkSecureRandomProvider)
+                || isNullOrBlank(restricts.jdkSecureRandomAlgorithm)) {
+            printStackTraceAndExit("Restricted security mode secure random is null.");
+        }
+    }
+
+    /**
+     * Check if restricted security policy sunset.
+     *
+     * @param descSunsetDate the sun set date from java.security
```

Comment:
The summary could be improved (missing verb).
Also 'sunset' is one word, not two.